src/docfx/build/xref/XrefMap.cs

```diff
@@ -11,22 +11,41 @@ internal class XrefMap
     {
         private readonly IReadOnlyDictionary<string, Lazy<ExternalXrefSpec>> _externalXrefMap;
         private readonly IReadOnlyDictionary<string, InternalXrefSpec> _internalXrefMap;
+        private readonly DependencyMapBuilder _dependencyMapBuilder;
 
-        public XrefMap(Context context, Docset docset, RestoreFileMap restoreFileMap)
+        public XrefMap(Context context, Docset docset, RestoreFileMap restoreFileMap, DependencyMapBuilder dependencyMapBuilder)
         {
             _internalXrefMap = InternalXrefMapBuilder.Build(context);
             _externalXrefMap = ExternalXrefMapLoader.Load(docset, restoreFileMap);
+            _dependencyMapBuilder = dependencyMapBuilder;
         }
 
-        public IXrefSpec Resolve(SourceInfo<string> uid)
+        public (Error, IXrefSpec) Resolve(SourceInfo<string> uid, Document referenceFile)
```

Comment:
How about Resolve<T> where T : IXrefSpec?

Reply:
Do not quite follow here, T is the return type?

Reply:
edited, the html tag is not showed :(

Reply:
I guess `T` would be used for the return type? What is the difference?

Reply:
The main problem here is, we need to provide two types of IXrefSpec, one for `DependencyResolver.ResolveXref` which only needs lazy properties, another one for `JsonSchemaTransformer` which needs properties loaded.

Reply:
What I means is to totally hide Internal Xref since you don't want others to directly use it expected DependencyResolver.